Resilio Connect API/JSMonitor/app.js

```diff
@@ -36,3 +36,41 @@ setTimeout(function() {console.log(enumerateAgents());}, 5000);
 
 console.log(findArrayDiff([1, 5, 17, 149, 150, 151, 152, 153, 154, 155, 156, 157, 158, 159, 160], [1, 5, 17, 150, 151, 152, 153, 154, 155, 156, 157, 158, 159, 160, 165]));
 
+var prevList;
+var callCounter = 0;
+
+class updateListofAgents {
+
+    constructor() {
+        this.updatedList = [];
+        this.arrayDiff = [];
+    }
+    
+    periodicUpdate(freq, ifDifferent) {
+        prevList = this.updatedList;
+        this.updatedList = enumerateAgents();
+
+        this.arrayDiff = findArrayDiff(this.updatedList, prevList);
+        if (callCounter != 0) {
+            if (this.arrayDiff.length != 0) {
+                ifDifferent(this.arrayDiff);
+            } else {
+                console.log("ALERT: No new agents this cycle");
+            }
+        }
+        updateAgentList();
+        callCounter++;
+        setTimeout(function() {update.periodicUpdate(freq, ifDifferent);}, freq);
+    }
+
+}
+
+const update = new updateListofAgents()
+update.periodicUpdate(30000, (diffArray) => {
+    if (callCounter == 1) {
+        console.log("ALERT: The initial agents are " + diffArray)
+    } else {
+        console.log("ALERT: The new agents this cycle are " + diffArray);
```

Comment:
The callback function should not use "callCounter" which is an internal variable.  It should use the value that is passed to it which is the diffArray.